contracts/core/src/octane/Staking.sol

```diff
@@ -230,6 +238,18 @@ contract Staking is OwnableUpgradeable, EIP712Upgradeable {
         _delegate(delegator, validator);
     }
 
+    /**
+     * @notice Undelegate tokens from a validator
+     * @dev Proxies x/staking.MsgUndelegate
+     * @param validator The address of the validator to undelegate from
+     * @param undelegationAmount The amount of ether tokens to undelegate
+     */
+    function undelegate(address validator, uint256 undelegationAmount) external payable {
```

Comment:
nit: rename to `amount` since `undelegationAmount` stutters a bit, it is the only amount and the context is clear. This also aligns with your emitted event nameing.